src/Agent.Listener/Diagnostics/MtuInfo.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Net.NetworkInformation;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Diagnostics
+{
+    class MtuInfo : IDiagnostic
+    {
+        public bool Execute(ITerminal terminal)
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            IPGlobalProperties properties = IPGlobalProperties.GetIPGlobalProperties();
+            terminal.WriteLine(string.Format("IP interface information for {0}.{1}", properties.HostName, properties.DomainName));
+            terminal.WriteLine();
+
+            foreach (NetworkInterface adapter in nics)
+            {
+                terminal.WriteLine(adapter.Description);
+
+                if (adapter.Supports(NetworkInterfaceComponent.IPv4))
+                {
+                    IPInterfaceProperties adapterProperties = adapter.GetIPProperties();
+                    IPv4InterfaceProperties p = adapterProperties.GetIPv4Properties();
```

Comment:
shocking IPv4InterfaceProperties  & IPv6InterfaceProperties, don't extend the same interface which is why I didn't combine these into 1 function and call it with different options.  Shocking!